lib/chef/mixin/windows_architecture_helper.rb

```diff
@@ -43,6 +43,14 @@ def wow64_architecture_override_required?(node, desired_architecture)
       end
 
       def with_os_architecture(node)
+        node ||= begin
```

Comment:
there are places where I do not have or ohai data. See the dsc resource cache. It is a singleton class